### PR TITLE
Some cleanup

### PR DESCRIPTION
**Pull request by Razican**
This Pull Request cleans up a bit the repositoy, especially regarding the WASM demo.

It changes the following:
 - Adds [Prettier][prettier] formatting
 - Adds checks in the CI for both, the JS/MD/YML styles and the fact that the WASM target builds.
 - Upgrades dependencies for both the Rust crates and the boa WASM demo.
 - Removes the `Makefile` that we were not using.
 - Adds `#[track_caller]` in the parser test functions, so that we know what test snippet is failing, instead of all panics pointing to the checker functions. This is a new feature of Rust 1.46, so this raises the minimum supported Rust version.

[prettier]: https://prettier.io/

### CODE REVIEW

**Comment by Razican** [comment]
It seems that the wasm-pack build is failing, but can't figure out why.

**Review by HalidOdat — COMMENTED** [review]
This looks good! :) maybe we should leave the wasm build out for now until we discover why its not building.

What do you think?

**Comment by Razican** [comment]
> This looks good! :) maybe we should leave the wasm build out for now until we discover why its not building.
> 
> What do you think?

I would like to do some extra debugging during this week, before merging.

**Comment by Razican** [comment]
Fixed by forcing the `wasm-pack` path :)

File: boa/benches/bench_scripts/array_pop.js [hunk 1]
```diff
@@ -1,24 +1,221 @@
...
-                     27, 29, 2828, 234, 23, 56, 32, 56, 67, 77, 32, 45, 93,
-                     17, 28, 83, 62, 99, 36, 28, 93, 27, 29, 2828, 234, 23,
-                     56, 32, 56, 67, 77, 32, 45, 93, 17, 28, 83, 62, 99, 36,
-                     28, 93, 27, 29, 2828, 234, 23, 56, 32, 56, 67, 77, 32,
-                     45, 93, 17, 28, 83, 62, 99, 36, 28, 93, 27, 29, 2828, 234,
-                     23, 56, 32, 56, 67, 77, 32, 45, 93, 17, 28, 83, 62, 99,
-                     36, 28, 93, 27, 29, 2828, 234, 23, 56, 32, 56, 67, 77, 32,
-                     45, 93, 17, 28, 83, 62, 99, 36, 28];
+(function () {
+  let testArray = [
```
Comment (Razican) [inline boa/benches/bench_scripts/array_pop.js:2]: I am not 100% sure of this array formatting, but I guess it's the standard JS formatting.

**Review by HalidOdat — APPROVED** [review]

**Pull request merged**